buildflow/api/providers.py

```diff
@@ -0,0 +1,36 @@
+import dataclasses
+from enum import Enum
+from typing import Any, Dict
+
+
+class Provider:
```

Comment:
Thoughts on adding a provider cache?

Thinking of the case where someone defines the same postgres provider multiple times across different files, ideally that would only keep one reference